xmtp_mls/src/builder.rs

```diff
@@ -295,6 +295,76 @@ mod tests {
         }
     }
 
+    // First, create a client1 using legacy key and then test following cases:
+    // - create client2 from same db with [IdentityStrategy::CachedOnly]
+    // - create client3 from same db with [IdentityStrategy::CreateIfNotFound]
+    // - create client4 with different db.
+    #[tokio::test]
+    async fn test_2nd_time_client_creation() {
```

Comment:
Will this test fail if you run it 2X against the same backend without clearing the backend's DB?

I would expect it to, since the first time it will register using the legacy key and publish an `IdentityUpdate`. The second time it will attempt to register the same inbox_id using the legacy key, which will cause `client_1` to behave like `client_4`, which _does_ require a signature request.

Reply:
Ideally we would have a helper function that could generate new legacy keys for tests

Reply:
Ya, and other tests seem failing because using the same `legacy_account_address`, this can be solved by generating random legacy_key & legacy_account_address.

Reply:
Added `generate_random_legacy_key` for this